### PR TITLE
NOS-519/Migrate to androidx

### PR DESCRIPTION
**Pull request by ouchadam**
Replaces the `support library` imports with `androidx`

- Upgrades room to the latest version

### CODE REVIEW

**Comment by ouchadam** [comment]
this work enables #521 

https://github.com/ouchadam/download-manager/compare/NOS-519-migrate-to-androidx...ouchadam:NOS-521-workmanager?expand=1

**Review by zegnus — DISMISSED** [review]
super 💯

**Comment by Mecharyry** [comment]
test this

File: gradle.properties [hunk 1]
```diff
@@ -15,3 +15,5 @@ org.gradle.jvmargs=-Xmx1536m
 # This option should only be used with decoupled projects. More details, visit
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
+android.useAndroidX=true
+android.enableJetifier=true
```
Comment (ouchadam) [inline gradle.properties:19]: we need this for the evernote library but it'll be removed soon!

**Comment by Mecharyry** [comment]
test this

**Comment by hal9002** [comment]
Can one of the admins review this PR?

**Comment by ouchadam** [comment]
hmmmm 🤔 any CI reports I'm able to view?

**Comment by MechaRyRy** [comment]
Sorry @ouchadam for some reason you can't see them on here when the PR is closed. I've changed the config so that it won't close PRs when they fail now.

**Comment by ouchadam** [comment]
Looks like there's some pmd errors, will fix

**Comment by ouchadam** [comment]
this PR failed because there's a newer version `okHttp` 🤦‍♂

**Review by MechaRyRy — APPROVED** [review]

**Comment by Mecharyry** [comment]
test this hal 😂

**Pull request merged**